lib/cc/engine/analyzers/issue.rb

```diff
@@ -0,0 +1,89 @@
+require "cc/engine/analyzers/sexp_lines"
+require "cc/engine/analyzers/violation_read_up"
+require "digest"
+
+module CC
+  module Engine
+    module Analyzers
+      class Issue
+        def initialize(language_strategy:, check_name:, current_sexp:, other_sexps:)
+          @language_strategy = language_strategy
+          @check_name = check_name
+          @current_sexp = current_sexp
+          @other_sexps = other_sexps
+        end
+
+        def format # rubocop:disable Metrics/MethodLength
+          {
+            "type": "issue",
+            "check_name": check_name,
+            "description": description,
+            "categories": ["Duplication"],
+            "location": format_location,
+            "remediation_points": calculate_points,
+            "other_locations": format_other_locations,
+            "content": content_body,
+            "fingerprint": fingerprint,
+          }
+        end # rubocop:enable Metrics/MethodLength
+
+        def report_name
+          "#{current_sexp.file}-#{current_sexp.line}"
+        end
+
+        def mass
+          current_sexp.mass
+        end
+
+        private
+
+        attr_reader :language_strategy, :check_name, :other_sexps, :current_sexp
+
+        def calculate_points
+          language_strategy.calculate_points(mass)
+        end
+
+        def format_location
+          format_sexp(current_sexp)
+        end
+
+        def format_other_locations
+          other_sexps.map do |sexp|
+            format_sexp(sexp)
+          end
+        end
+
+        def format_sexp(sexp)
+          lines = SexpLines.new(sexp)
+          {
+            "path": sexp.file.gsub(%r{^./}, ""),
+            "lines": {
+              "begin": lines.begin_line,
+              "end": lines.end_line,
+            },
+          }
+        end
+
+        def content_body
+          { "body": ViolationReadUp.new(mass).contents }
+        end
+
+        def fingerprint
+          digest = Digest::MD5.new
+          digest << current_sexp.file
```

Comment:
I know this has been discussed to death here, so pardon me for bringing it up again, but I'm still not 100% clear on what we decided the right thing to do here is.

This implementation will result in the fingerprint of two duplication issues in the same file (even duplication of different pieces of code that happen to have the same mass) having the same fingerprint. You're saying that this collection of data (file name, mass, check name) was how classic calculated the fingerprint, and such a collision is expected?

Since our comparison algorithm relies on fingerprints, I'm pretty sure this could result in an analysis producing new duplication issues, but the PR would still show "no new issues" because of the fingerprint collision. That does not sound like desired behavior to me, so I'm still not convinced that "do what classic did" is the right call here.